stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -670,13 +670,31 @@ function opnorm1(A::AbstractMatrix{T}) where T
     return convert(Tnorm, nrm)
 end
 
+# Uses power iteration to compute the maximal singular value.
+# falls back to svdvals if it runs into numerics issues.
 function opnorm2(A::AbstractMatrix{T}) where T
     require_one_based_indexing(A)
     m,n = size(A)
     Tnorm = typeof(float(real(zero(T))))
     if m == 0 || n == 0 return zero(Tnorm) end
     if m == 1 || n == 1 return norm2(A) end
-    return svdvals(A)[1]
+    # to minimize the chance of x being orthogonal to the largest eigenvector
+    x = randn(Tnorm, n)
```

Comment:
This is the wrong type — it should be `typeof(float(zero(T))`, or equivalently `float(T)`, since in general the eigenvectors will be complex if `T` is complex.